lib/models/LifecycleConfiguration.js

```diff
@@ -0,0 +1,623 @@
+            filter: {
+                rulePrefix: <value>,
+                tags: [
+                    {
+                        key: <value>,
+                        val: <value>
+                    },
+                    {
+                        key: <value>,
+                        val: <value>
+                    }
+                ]
+            },
+            actions: [
+                {
+                    actionName: <value>,
+                    days: <value>,
+                    date: <value>,
+                    deleteMarker: <value>
+                },
+                {
+                    actionName: <value>,
+                    days: <value>,
+                    date: <value>,
+                    deleteMarker: <value>,
+                },
+            ]
+        }
+    ]
+  };
+ */
+
+class LifecycleConfiguration {
+    /**
+     * Create a Lifecycle Configuration instance
+     * @param {string} xml - the parsed xml
+     * @return {object} - LifecycleConfiguration instance
+     */
+    constructor(xml) {
+        this._parsedXML = xml;
+        this._ruleIDs = [];
+        this._config = {};
+    }
+
+    /**
+     * Get the lifecycle configuration
+     * @return {object} - the lifecycle configuration
+     */
+    getLifecycleConfiguration() {
+        const rules = this._buildRulesArray();
+        if (rules.error) {
+            this._config.error = rules.error;
+        }
+        return this._config;
+    }
+
+    /**
+     * Build the this._config.rules array
+     * @return {object} - contains error if any rule returned an error
+     * or parsing failed
+     */
+    _buildRulesArray() {
+        const rules = {};
+        this._config.rules = [];
+        if (!this._parsedXML || this._parsedXML === '') {
+            rules.error = errors.MalformedXML.customizeDescription(
+                'request xml is undefined or empty');
+            return rules;
+        }
+        if (!this._parsedXML.LifecycleConfiguration &&
+            this._parsedXML.LifecycleConfiguration !== '') {
+            rules.error = errors.MalformedXML.customizeDescription(
+                'request xml does not include LifecycleConfiguration');
+            return rules;
+        }
+        const lifecycleConf = this._parsedXML.LifecycleConfiguration;
+        const rulesArray = lifecycleConf.Rule;
+        if (!rulesArray || rulesArray.length === 0) {
+            rules.error = errors.MalformedXML.customizeDescription(
+                'request xml does not include at least one rule');
+            return rules;
+        }
+        if (rulesArray.length > 1000) {
+            rules.error = errors.MalformedXML.customizeDescription(
+                'request xml includes over max limit of 1000 rules');
+            return rules;
+        }
+        for (let i = 0; i < rulesArray.length; i++) {
+            const rule = this._parseRule(rulesArray[i]);
+            if (rule.error) {
+                rules.error = rule.error;
```

Comment:
If I'm not mistaken, we could return immediately here because we will send an error response and don't care about the remaining rules.